modules/aggregation-layers/src/contour-layer/marching-squares.js

```diff
@@ -162,12 +189,32 @@ export function getVertices(opts) {
 
   // default case is ISO_LINES
   const lines = [];
-  offsets.forEach(xyOffsets => {
-    xyOffsets.forEach(offset => {
-      const vX = refVertexX + offset[0] * cellSize[0];
-      const vY = refVertexY + offset[1] * cellSize[1];
+  offsets.forEach((xyOffsets, i) => {
+    xyOffsets.forEach((offset, j) => {
+      const li = getLinearOffsetValue(linearOffsets[i][j], weights, thresholdData.threshold);
+      let xOffset;
+      let yOffset;
+      if (isLI) {
+        xOffset = !offset[0] ? offset[0] + li : offset[0];
+        yOffset = !offset[1] ? offset[1] + li : offset[1];
+      } else {
+        xOffset = offset[0];
+        yOffset = offset[1];
+      }
+      const vX = refVertexX + xOffset * cellSize[0];
+      const vY = refVertexY + yOffset * cellSize[1];
       lines.push([vX, vY, vZ]);
     });
   });
   return lines;
 }
+
+function getLinearOffsetValue(offset, weights, threshold) {
```

Comment:
I would recommend merging `getLinearInterpolation` method into `getLienarOffsetValue`, and change function name to something like ,  `getsmoothOffset` or just `getOffset` which will internally check if interpolation is on/off and return the correct offset.

Reply:
I will merge these two methods. 

Before merging the PR we are removing the interpolation check, right?

Reply:
If we have vertex generation test cases, and if we validate all cases using unit tests, we can make the prop experimental and land the PR. Effectively all this new functionality is enabled only when this prop is set.